src/loki/matrix_action.cc

```diff
@@ -52,8 +66,17 @@ namespace {
           LOG_DEBUG("max_location_distance -> " + std::to_string(max_location_distance));
         }
 
-        if (path_distance > matrix_max_distance)
-          throw valhalla_exception_t{400, 154};
+        if (path_distance > matrix_max_distance) {
+          rapidjson::Value pair(rapidjson::kObjectType);
+          pair.AddMember("source", s, request.GetAllocator());
+          pair.AddMember("target", t, request.GetAllocator());
+
+          ignore_list->PushBack(pair,request.GetAllocator());
+          invalid_count++;
+
+          if ((static_cast<float>(invalid_count) / total_count) > kPercentInvalidAllowed)
```

Comment:
rather than doing a division at each stage, I would compute a max invalid count outside the loop and compare to that